src/dataset/infrastructure/mappers/JSDatasetMapper.ts

```diff
@@ -29,19 +35,22 @@ export class JSDatasetMapper {
         jsDataset.alternativePersistentId,
         jsDataset.publicationDate,
         jsDataset.citationDate
-      )
+      ),
+      privateUrl
     ).build()
   }
 
   static toVersion(
     jDatasetVersionId: number,
-    jsDatasetVersionInfo: JSDatasetVersionInfo
+    jsDatasetVersionInfo: JSDatasetVersionInfo,
+    requestedVersion?: string
```

Comment:
There are some e2e tests that are checking the DatasetVersion that need to be fixed because of the added requestedVersion property